File: Data Structures/simpleLinkedList.c
#include <stdio.h>
#include <stdlib.h>

struct Node {
	int data;          // data contained in the node
	struct Node* next; // pointer to the next Node element
};

void printList(struct Node* n){
	while (n != NULL){
		printf(" %d ", n->data);
		n = n-> next;
	}
}

int main(){
	struct Node* first = NULL;
	struct Node* second = NULL;
	struct Node* third = NULL;
	
	// Create the space for the 3 nodes in the heap
	first = (struct Node*)malloc(sizeof(struct Node));
	second = (struct Node*)malloc(sizeof(struct Node));
	third = (struct Node*)malloc(sizeof(struct Node));

	first->data = 1; 
	first->next = second;

	second->data = 2;
	second->next = third;

	third->data = 3;
	third->next = NULL;

	// Print the list
	printList(first);

	return 0;
}

File: Data Structures/simpleLinkedList.py
"""Simple Linked List."""

class Node:

    def __init__(self, data):
        self.data = data # Data
        self.next = None # Reference to the next element of the linked list

class LinkedList:

    def __init__(self):
        self.head = None # First element of the list

    def printList(self):
        """Prints the content of a linked list."""
        temporal = self.head
        while temporal:
            print(temporal.data)
            temporal = temporal.next 

if __name__ == "__main__":

    linkedL = LinkedList()  # Initizialize the list as empty

    linkedL.head = Node(1)  # Add the first element to the list

    second = Node(2)  # Add data of the second node 
    linkedL.head.next = second  # Include the second node as the next element in the list

    third = Node(3)  # Add data od the third node
    second.next = third  # Links the third node to the second one

    fourth = Node(4)  # And so on...
    third.next = fourth
    
    linkedL.printList()


